TrafficCapture/trafficCaptureProxyServer/src/main/java/org/opensearch/migrations/trafficcapture/proxyserver/netty/ProxyChannelInitializer.java

```diff
@@ -0,0 +1,58 @@
+package org.opensearch.migrations.trafficcapture.proxyserver.netty;
+
+import io.netty.channel.ChannelInitializer;
+import io.netty.channel.socket.SocketChannel;
+import io.netty.handler.codec.http.DefaultHttpRequest;
+import io.netty.handler.codec.http.HttpMethod;
+import io.netty.handler.codec.http.HttpObjectDecoder;
+import io.netty.handler.logging.LogLevel;
+import io.netty.handler.logging.LoggingHandler;
+import io.netty.handler.ssl.SslHandler;
+import org.opensearch.migrations.trafficcapture.IConnectionCaptureFactory;
+import org.opensearch.migrations.trafficcapture.netty.ConditionallyReliableLoggingHttpRequestHandler;
+import org.opensearch.migrations.trafficcapture.netty.LoggingHttpRequestHandler;
+import org.opensearch.migrations.trafficcapture.netty.LoggingHttpResponseHandler;
+
+import javax.net.ssl.SSLEngine;
+import java.io.IOException;
+import java.util.function.Supplier;
+
+public class ProxyChannelInitializer extends ChannelInitializer<SocketChannel> {
+
+    private final IConnectionCaptureFactory connectionCaptureFactory;
+    private final Supplier<SSLEngine> sslEngineProvider;
+    private final String host;
+    private final int port;
+
+    public ProxyChannelInitializer(String host, int port, Supplier<SSLEngine> sslEngineSupplier,
+                                   IConnectionCaptureFactory connectionCaptureFactory) {
+        this.host = host;
+        this.port = port;
+        this.sslEngineProvider = sslEngineSupplier;
+        this.connectionCaptureFactory = connectionCaptureFactory;
+    }
+
+    public boolean shouldGuaranteeMessageOffloading(DefaultHttpRequest httpRequest) {
+        return (httpRequest != null &&
+                (httpRequest.method().equals(HttpMethod.POST) ||
+                        httpRequest.method().equals(HttpMethod.PUT) ||
+                        httpRequest.method().equals(HttpMethod.DELETE)));
+    }
+
+    @Override
+    protected void initChannel(SocketChannel ch) throws IOException {
+        var sslContext = sslEngineProvider != null ? sslEngineProvider.get() : null;
+        if (sslContext != null) {
+            ch.pipeline().addLast(new SslHandler(sslEngineProvider.get()));
```

Comment:
Assuming the SslHandler decrypts the incoming data for the following handlers, do we do any re-encryption before we send to the source cluster?